### PR TITLE
Feature/signatures

### PR DESCRIPTION
**Pull request by glow-mdsol**
@isparks @rsayer-mdsol @anewbigging 
* @rsayer-mdsol amended release to add new elements 
* amended sphinx to find docstrings
* added docstrings to builders
* added LRP type from schema and used for Checks and Derivations
* Added support for python 3-6

### CODE REVIEW

File: rwslib/builders.py [hunk 1]
```diff
@@ -168,65 +250,57 @@ def build(self, builder):
...
+        :param date_time_stamp: :class:`DateTimeStamp` for the time of Signature
+        """
         self._id = None
-        self.id = identifier
-        self.include_file_oid = include_file_oid
-        self.user_ref = None
-        self.location_ref = None
-        self.reason_for_change = None
-        self.date_time_stamp = None
+        self.id = id
```
Comment (isparks) [inline rwslib/builders.py]: Use of 'id' here shadows pythons id function.

File: rwslib/builders.py [hunk 2]
```diff
@@ -235,56 +309,345 @@ def build(self, builder):
...
+                self << flags
+            else:
+                raise AttributeError("Flags attribute should be an iterable or Flag")
+        self._id = None
+        if id is not None:
+            self.id = id
+        self._seqnum = None
+        if seqnum is not None:
+            # validate the input
+            self.seqnum = seqnum
```
Comment (isparks) [inline rwslib/builders.py]: self._seqnum = seqnum?

File: rwslib/builders.py (hunk 1 above)
Comment (glow-mdsol) [inline rwslib/builders.py]: Ok, @rsayer-mdsol one we should tidy up

File: rwslib/builders.py (hunk 2 above)
Comment (glow-mdsol) [inline rwslib/builders.py]: We use the setter below to allocate the value

File: rwslib/builders.py [hunk 3]
```diff
@@ -335,6 +703,15 @@ class ItemData(TransactionalElement):
     ALLOWED_TRANSACTION_TYPES = ['Insert', 'Update', 'Upsert', 'Context', 'Remove']
 
     def __init__(self, itemoid, value, specify_value=None, transaction_type=None, lock=None, freeze=None, verify=None):
+        """
+        :param str itemoid: OID for the matching :class:`ItemDef`
+        :param str value: Value for the the ItemData
+        :param str specify_value: 'If other, specify' value - *Rave specific attribute*
+        :param str transaction_type: Transaction type for the data
+        :param bool lock: Locks the DataPoint? - *Rave specific attribute*
```
Comment (isparks) [inline rwslib/builders.py]: Question marks here. Is there doubt?

**Comment by isparks** [comment]
Comments addressed. Nice work!

**Pull request merged**